harness/determined/profiler.py

```diff
@@ -100,17 +100,20 @@ def __init__(
 
         # If the ProfilingAgent is disabled, don't waste resources by creating useless threads
         if self.is_enabled:
-            # Set up timer thread to stop collecting after 5 minutes
-            self.max_collection_seconds = 300
-            self.shutdown_timer = PreemptibleTimer(
-                self.max_collection_seconds, self._end_collection
-            )
-
-            # Set up the thread responsible for making API calls
+            # Set up sender thread with the number of producers to expect shutdowns from.
             self.send_queue = (
                 queue.Queue()
             )  # type: """queue.Queue[Union[List[TrialProfilerMetricsBatch], ShutdownMessage]]"""
-            self.sender_thread = ProfilerSenderThread(self.send_queue, self.master_url)
+
+            # Set up timer thread to stop collecting after 5 minutes
+            self.shutdown_timer = PreemptibleTimer(MAX_COLLECTION_SECONDS, self._end_collection)
+
+            # Set up the thread responsible for making API calls
+            if self.sysmetrics_is_enabled or self.timings_is_enabled:
```

Comment:
This check is already handled by `if self.is_enabled` (`is_enabled` will be `False` if both `sysmetrics_is_enabled` and `timings_is_enabled` are `False`). However, I forgot that it worked like that when I wrote my original comment and you didn't notice it when you were writing the code, so maybe it is unintuitive and should be changed. Or just drop this `if`

Reply:
kinda just reordered stuff to make it look better/more logical